test/result/combine.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { Result } from '@/src/result';
+
+describe('Result', () => {
+  describe('combine', () => {
+    test('fails if one result fails', () => {
+      const success = Result.success(1);
+      const failure = Result.failure('1st Error');
+
+      const result = Result.combine({ success, failure });
+
+      expect(result.isFailure).toBe(true);
+    });
+
+    test('succeeds if one results succeed', () => {
+      const success = Result.success(1);
+
+      const result = Result.combine({ success });
+
+      expect(result.isSuccess).toBe(true);
+    });
+
+    test('succeeds if all results succeed', () => {
+      const success_1 = Result.success(1);
+      const success_2 = Result.success(2);
+
+      const result = Result.combine({ success_1, success_2 });
+
+      expect(result.isSuccess).toBe(true);
+    });
+
+    test('yields all result values on success', () => {
+      const success_1 = Result.success(1);
+      const success_2 = Result.success({ name: 'Arthur' });
+
+      const result = Result.combine({ success_1, success_2 });
+
+      const values = result.getValueOrThrow();
+
+      expect(values.success_1).toBe(1);
+      expect(values.success_2).toEqual({ name: 'Arthur' });
+    });
+
+    test('concatenates error messages', () => {
+      const failure_1_message = '1st Error';
```

Comment:
👍👍 Solid tests. I jumped straight to them to understand how `combine()` was supposed to work.

There are custom assertions in `/test/expectExtensions.ts` (you can see how they're used in `/test/result/map.spec.ts`, but you don't have to use them if you don't want to.